packages/boba/contracts/contracts/LP/L1LiquidityPool.sol

```diff
@@ -474,6 +488,47 @@ contract L1LiquidityPool is CrossDomainEnabledFast, ReentrancyGuardUpgradeable,
         );
     }
 
+    function clientDepositL1Batch(
+        ClientDepositToken[] calldata _tokens
+    )
+        external
+        payable
+        whenNotPaused
+    {
+        require(_tokens.length < 5, "Too Many Tokens");
+        uint256 ETHAmount;
+        ClientPayToken[] memory payload = new ClientPayToken[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            ClientDepositToken memory token = _tokens[i];
+            PoolInfo storage pool = poolInfo[token.l1TokenAddress];
+            if (token.l1TokenAddress != address(0)) {
+                IERC20(token.l1TokenAddress).safeTransferFrom(msg.sender, address(this), token.amount);
+            } else {
+                ETHAmount = token.amount;
+            }
+            payload[i] = ClientPayToken(msg.sender, pool.l2TokenAddress, token.amount);
+        }
+        require(ETHAmount == msg.value, "Invalid ETH Amount");
```

Comment:
this require would only verify the last token amount? is this correct?

Reply:
The last one is to verify that the msg.value is equal to the amount that the user provides in the payload. 
```
  const tx = await Proxy__L1LiquidityPool.clientDepositL1Batch([
    {
      amount: ethers.utils.parseEther('1'),
      l1TokenAddress: ethers.constants.AddressZero
    }
  ], {value: ethers.utils.parseEther('1')})

```